lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +196,41 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.config = config
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        current_url = article_bs.find('meta', property='og:url').get('content')
+        for link in article_bs.find_all('a',
+                                        class_=lambda value: value and ('mininews' in value or 'midinews' in value)):
+            href = link.get('href')
+            if href:
+                abs_href = urljoin(current_url, href)
+                return abs_href
+        return None
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        for seed_url in self.get_search_urls():
+            response = make_request(seed_url, self.config)
+            if response.status_code == 200:
+                html = response.text
+                main_bs = BeautifulSoup(html, 'html.parser')
+                urls = self._extract_url(main_bs)
+                self.urls.append(urls)
```

Comment:
you do not control number of elements to collect?

Reply:
corrected